--- a/src/r8125_n.c
+++ b/src/r8125_n.c
@@ -15190,7 +15190,6 @@
 #else
         if (tp->HwSuppRssVer > 0 && tp->HwCurrIsrVer > 1) {
-                u16 rss_queue_num = netif_get_num_default_rss_queues();
-                tp->num_rx_rings = min(tp->HwSuppNumRxQueues, rss_queue_num);
+                tp->num_rx_rings = min(tp->HwSuppNumRxQueues, (u16)4);
                 if (!(tp->num_rx_rings >= 2 && tp->irq_nvecs >= tp->num_rx_rings))
                         tp->num_rx_rings = 1;
